qmi/core/transport.py

```diff
@@ -647,18 +630,38 @@ def read(self, nbytes: int, timeout: Optional[float]) -> bytes:
         while nbuf < nbytes:
             self._safe_socket.settimeout(tremain)
             try:
-                b = self._safe_socket.recv(nbytes - nbuf)
+                if self._assert_addr:
+                    # For UDP, reading data fails if the arriving packet size is larger than the given nbytes (- nbuf)
+                    # to read. To avoid this happening, read at least 4096 standard that is larger than regular MTU.
+                    udp_packet_size = max(nbytes - nbuf, 4096)
+                    try:
+                        b, addr = self._safe_socket.recvfrom(udp_packet_size)
+                    except (BlockingIOError, socket.timeout) as err:
+                        raise err  # Re-raise this for handling in the upper try-except.
+                    except OSError:
+                        raise QMI_RuntimeException(f"UDP packet size was larger than {udp_packet_size}. Data is lost.")
+
+                    if addr != self._address:
+                        _logger.warning(f"Received data from address %s while expected data only from %s!", addr,
```

Comment:
you are using an f-string but are providing % operator. Since we are logging we can use lazy formatting. This should work:
`_logger.warning("Received data from address %s while expected data only from %s!", addr,
                                        self._address)`

Reply:
Good catch, will be fixed!

Reply:
This got removed in the end.